src/Dispatcher.cpp

```diff
@@ -83,7 +88,7 @@ void DispatcherBase::StartServer(
     if (m_active) return;
     m_active = true;
   }
-  m_server = true;
+  m_networkMode = NT_NET_MODE_SERVER;
```

Comment:
This shouldn't be set here, as this results in a race condition. The mode would say server, even though at this point the server might not have actually started. Put these in the individual threads after Start() (server) and Accept() (client) have successfully returned.